lib/wab/impl/model.rb

```diff
@@ -127,22 +125,19 @@ def update(obj, rid, where, filter)
         { code: 0, updated: updated }
       end
 
-      def delete(del_opt, rid, where, filter)
+      def delete(_del_opt, _rid, where, filter)
         deleted = []
         @lock.synchronize {
           if where.is_a?(Expr)
             @map.each { |ref,obj|
-              if where.eval(obj) && (filter.nil? || filter.eval(obj))
-                deleted << ref
-                @map.delete(ref)
-              end
+              next unless where.eval(obj) && (filter.nil? || filter.eval(obj))
+              deleted << ref
+              @map.delete(ref)
```

Comment:
Don't see how this improves readability or understanding.

Reply:
`if..else` inside an `each` block.. shouldn't that have an impact on readability / performance?

Reply:
There is no performance difference unless the guard is more expensive but I doubt there is any at all. As for readability, most people read code similar to how they talk. So as an example, which reads more smoothly and is easier to understand.

1) For each person in the room, if they are wearing a hat, smile.
2) For each person in the room, if they are not wearing a hat continue to the next person. Still here, smile.

Reply:
obviously option 1.. :smiley: 
that's a wonderful example to illustrate readability..

Reply:
😄

Reply:
The example scenario can also be defined as:
  **Looking at each person wearing a hat and smiling..**
  *Look at each person's face.*
  *Next face if there's no hat on the current head*
  *Smile*

Reply:
That said, if there's no performance gain from using the `next`, I'll revert..

Reply:
Something like that, yes. First one is easy to follow. Not always that case if there are multiple conditions but for simple yes/no if is easier IMHO.